go-concurrency-exercises/main.go

```diff
@@ -0,0 +1,64 @@
+
+//////////////////////////////////////////////////////////////////////
+//
+// Given is a producer-consumer scenario, where a producer reads in
+// tweets from a mockstream and a consumer is processing the
+// data. Your task is to change the code so that the producer as well
+// as the consumer can run concurrently
+//
+
+package main
+
+import (
+	"fmt"
+	"go-concurrency-exercises/mockstream"
+	"time"
+)
+
+func producer(stream mockstream.Stream, tweetChan chan *mockstream.Tweet, doneChan chan bool) {
+	defer close(tweetChan)
+	for {
+		tweet, err := stream.Next()
+		if err == mockstream.ErrEOF {
+			doneChan <- true
+			return
+		}
+		tweetChan <- tweet
+	}
+}
+
+func consumer(tweetChan chan *mockstream.Tweet, doneChan chan bool) {
+	for {
+		select {
+		case tweet, ok := <-tweetChan:
+			if ok {
+				if tweet.IsTalkingAboutGo() {
+					fmt.Println(tweet.Username, "\ttweets about golang")
+				} else {
+					fmt.Println(tweet.Username, "\tdoes not tweet about golang")
+				}
+			} else {
+				return
+			}
+		case <-doneChan:
```

Comment:
you don't need that channel. just try to remove it at all and then start the app